File: frontend/src/api/knowledge.ts
import request from './request'

export interface KnowledgeEntry {
  id: number
  keyword: string
  title: string
  content: string
  source_url?: string
  category?: string
  usage_count: number
  created_at: string
}

export const knowledgeApi = {
  search: (keyword: string, maxResults = 3, useAI = false) =>
    request.post<KnowledgeEntry[]>('/knowledge/search', {
      keywords: [keyword],
      max_results_per_keyword: maxResults,
      use_ai: useAI
    }),

  list: (keyword?: string) =>
    request.get<KnowledgeEntry[]>('/knowledge/', { params: { keyword } }),

  create: (data: { keyword: string; title: string; content: string; category?: string }) =>
    request.post<KnowledgeEntry>('/knowledge/', data),

  update: (id: number, data: { keyword?: string; title?: string; content?: string; category?: string }) =>
    request.put<KnowledgeEntry>(`/knowledge/${id}`, data),

  delete: (id: number) =>
    request.delete(`/knowledge/${id}`),
}

File: backend/app/schemas/knowledge.py
"""
知识库Schema
"""
from datetime import datetime
from typing import Optional, List
from pydantic import BaseModel


class KnowledgeEntryCreate(BaseModel):
    keyword: str
    title: str
    content: str
    source_url: Optional[str] = None
    category: Optional[str] = None
    tags: Optional[str] = None


class KnowledgeEntryResponse(BaseModel):
    id: int
    keyword: str
    title: str
    content: str
    source_url: Optional[str] = None
    source_type: str
    category: Optional[str] = None
    tags: Optional[str] = None
    char_count: int
    usage_count: int
    created_at: datetime
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class KnowledgeSearchRequest(BaseModel):
    keywords: List[str]
    max_results_per_keyword: int = 3
    use_ai: bool = False  # 是否使用AI增强搜索
